### PR TITLE
Input validation risdev 3801

### PR DESCRIPTION
**Pull request by hamo225**

### CODE REVIEW

File: frontend/src/components/controls/RisDateInput.vue [hunk 1]
```diff
@@ -117,17 +156,48 @@ watch(inputCompleted, (is) => {
...
-    placeholder="TT.MM.JJJJ"
-    :readonly="isReadOnly"
-    @blur="onBlur"
-    @keydown.delete="backspaceDelete"
-    @maska="onMaska"
-  />
+  <div
+    :class="
+      props.labelPosition === 'left'
+        ? `|| 'gap-8'} flex items-center gap-4`
```
Comment (malte-laukoetter) [inline frontend/src/components/controls/RisDateInput.vue]: it feels like there is missing some logic here (in regards to the `|| 'gap-8'}`)

File: frontend/src/components/controls/RisDateInput.vue [hunk 2]
```diff
@@ -117,17 +156,48 @@ watch(inputCompleted, (is) => {
...
+        : 'grid gap-2'
+    "
+  >
+    <!-- eslint-disable-next-line vuejs-accessibility/label-has-for -->
```
Comment (malte-laukoetter) [inline frontend/src/components/controls/RisDateInput.vue]: can we nest the input within this label? That should solve this warning.

File: frontend/src/components/controls/RisDateInput.vue [hunk 3]
```diff
@@ -4,23 +4,40 @@ import dayjs from "dayjs"
...
+    /** Label of the form field. */
+    label?: string
+    /** Validation error and message to display. */
+    validationError?: ValidationError
+    /** Label position: 'above' or 'left' */
+    labelPosition?: "above" | "left"
+    /** Custom classes for the container */
+    containerClass?: string
+    /** Custom classes for the label */
+    labelClass?: string
```
Comment (malte-laukoetter) [inline frontend/src/components/controls/RisDateInput.vue]: do we actually need this or is this just the same class list for every `RisDateInput` anyway?

**Review by malte-laukoetter — CHANGES_REQUESTED** [review]

File: frontend/src/components/controls/RisDateInput.vue (hunk 1 above)
Comment (hamo225) [reply frontend/src/components/controls/RisDateInput.vue]: Not needed anymore. Removing.

File: frontend/src/components/controls/RisDateInput.vue (hunk 2 above)
Comment (hamo225) [reply frontend/src/components/controls/RisDateInput.vue]: Good point

File: frontend/src/components/controls/RisDateInput.vue (hunk 3 above)
Comment (hamo225) [reply frontend/src/components/controls/RisDateInput.vue]: I have removed both.

**Pull request merged**